File: live-test/python/src/setter_super.py
"""Setter override across base/derived class hierarchy."""


class WithSetterBase:
    @property
    def value(self) -> str:
        return "base"

    @value.setter
    def value(self, v: str) -> None:
        pass


class WithSetter(WithSetterBase):
    @property
    def value(self) -> str:
        return "derived"

    @value.setter
    def value(self, v: str) -> None:
        pass

File: live-test/javascript/src/setter_super.js
'use strict';

class Base {
    set value(v) { this._v = v; }
    get value() { return this._v; }
}

class Derived extends Base {
    set value(v) { this._v = v.toUpperCase(); }
    get value() { return this._v; }
}

module.exports = { Base, Derived };
